public/js/counties.js

```diff
@@ -470,20 +470,29 @@ const countiesLocation = {
     center: '-0.56333,34.93583',
   },
 };
- 
-window.onload = function() {
-  const countySelect = document.getElementById("county-select");
-  countySelect.addEventListener("change",()=>{
-    var width = $(window).width(); 
-    if(width < 768 ){
-      $(".menu-button").click();
-    }})
-  Object.keys(countiesLocation).forEach(key => {
-  const c = new Option(countiesLocation[key].label, key);
-  // TODO(Brenda): We only have Nairobi data so we'll disable all other counties for now
-  if(c.value !== "nairobi" ){
-    c.disabled = true
-  }
-  countySelect.options.add(c);
+
+let results;
+
+window.onload = async () => {
+  const countySelect = document.getElementById('county-select');
+  countySelect.addEventListener('change', () => {
+    let width = $(window).width();
+    if (width < 768) {
+      $('.menu-button').click();
+    }
   });
-}; 
+  Object.keys(countiesLocation).forEach((key) => {
+    const c = new Option(countiesLocation[key].label, key);
+    // TODO(Brenda): We only have Nairobi data so we'll disable all other counties for now
+    if (c.value !== 'nairobi') {
+      c.disabled = true;
+    }
+    countySelect.options.add(c);
+  });
+
+  // fetch data. Blocking JS? + quite slow
+  const data = await fetch('/api/data/?length=7');
+  results = await data.json();
+
+  countyGraphChange('Nairobi');
```

Comment:
Shouldn't `results` be passed to `countyGraphChange`? Or how does it "know" where to get data?

Reply:
Order of JS files in the HTML file, result data is a variable that can be accessed in this scope @kilemensi

Reply:
I can see that @gertie-sheshe but when have we ever written function code that depends on global variables?

Reply:
Used a different approach @kilemensi Take a look